config/initializers/application_secrets.rb

```diff
@@ -0,0 +1,47 @@
+require './app/lib/secrets_manager'
+
+if Rails.env == 'production'
+  omniauth_secrets = SecretsManager.get_value('omniauth')
+  ak_secrets = SecretsManager.get_value('prod/actionKitApi')
+  database_secrets = SecretsManager.get_value('champaignDB')
+  share_progress_secrets = SecretsManager.get_value('shareProgressApi')
+  gocardless_secrets = SecretsManager.get_value('gocardless')
+  smtp_secrets = SecretsManager.get_value('prod/smtp')
+  mixpanel_secrets = SecretsManager.get_value('prod/mixpanel')
+  twilio_secrets = SecretsManager.get_value('prod/twilio')
+  devise_secrets = SecretsManager.get_value('deviseSecret')
+  key_base_secrets = SecretsManager.get_value('champaignSecretKeyBase')
+  recaptcha2_secrets = SecretsManager.get_value('recaptchaTwo')
+  call_targeting_secrets = SecretsManager.get_value('prod/callToolTargeting')
+  member_services_secrets = SecretsManager.get_value('memberServices')
+  champaign_api_secrets = SecretsManager.get_value('champaign')
+
+  ENV['OMNIAUTH_CLIENT_SECRET'] = omniauth_secrets['secret']
+  ENV['OMNIAUTH_CLIENT_ID'] = omniauth_secrets['clientId']
+  ENV['AK_USERNAME'] = ak_secrets['username']
+  ENV['AK_PASSWORD'] = ak_secrets['password']
+
+  ENV['RDS_DB_NAME'] = database_secrets['dbname']
+  ENV['RDS_USERNAME'] = database_secrets['username']
+  ENV['RDS_PASSWORD'] = database_secrets['password']
+  ENV['RDS_HOSTNAME'] = database_secrets['host']
+  ENV['RDS_PORT'] = database_secrets['port'].to_s
+
+  ENV['GOCARDLESS_TOKEN'] = gocardless_secrets['token']
+  ENV['GOCARDLESS_SECRET'] = gocardless_secrets['secret']
+
+  ENV['SHARE_PROGRESS_API_KEY'] = share_progress_secrets['apiKey']
+  ENV['SMTP_USERNAME'] = smtp_secrets['username']
+  ENV['SMTP_PASSWORD'] = smtp_secrets['password']
+  ENV['MIXPANEL_TOKEN'] = mixpanel_secrets['token']
+  ENV['TWILIO_ACCOUNT_SID'] = twilio_secrets['sid']
+  ENV['TWILIO_AUTH_TOKEN'] = twilio_secrets['token']
+  ENV['CALL_TARGETING_SECRET'] = call_targeting_secrets['secret']
+
+  ENV['DEVISE_SECRET_KEY'] = devise_secrets['secretKey']
+  ENV['SECRET_KEY_BASE'] = key_base_secrets['secretKeyBase']
+  ENV['CHAMPAIGN_API_KEY'] = champaign_api_secrets['apiKey']
+  ENV['RECAP2SIK'] = recaptcha2_secrets['siteKey']
+  ENV['RECAP2SEK'] = recaptcha2_secrets['secretKey']
+  ENV['MEMBER_SERVICES_SECRET'] = member_services_secrets['secret']
+end
```

Comment:
Looking at this I wondered if injecting 'secrets' directly into the settings config yml files would be possible. Looking at the github docs for the config gem I saw this: https://github.com/rubyconfig/config#working-with-aws-secrets-manager. From their example, it looks like you can build up the hash of key value pairs and then add them to settings with 

```ruby
secret_source = Config::Sources::EnvSource.new(secrets)
Settings.add_source!(secret_source)
Settings.reload!
```

Reply:
By adding them to the `Settings` object, it would allow us to continue using just the single API for accessing secrets

Reply:
I'll give it a try Omar 👍🏼

Reply:
I reviewed the docs from the link you provided; that code is not added on the yml file as it is not supported there.
But it can be added to this file I created. Will that work? I can do the commit and I can revert in case is not needed

Reply:
@osahyoun, check this commit and let me know what you think https://github.com/SumOfUs/Champaign/pull/1956/commits/922ee55a55d3c2ec33e3ede1e8dcaf6d99193909

Some things to mention:
1. Segments don't work (or if they do, I don't know how), for example, if you have this on the settings.yml:
```
gocardless:
    token: 'value'
```
I couldn't find a way to add it.

2. How do I add the database secrets in this way? I couldn't find any docs about it.